### ISSUE TITLE
Markdown tables can silently discard content containing `|`

### ISSUE THREAD

**Issue by kpreid** [rust-lang/rust#159186]
# Code
```rust
//! This is a table with some columns:
//!
//! | col1 | col2 | col3 | col4 |
//! | ---- | ---- | ---- | ---- |
//! | `one` | `two` | three | four |
//! | `code_with(|arg| arg)` |  `code_with(|arg| arg)` | three | four |
```

# Reproduction Steps
`cargo doc`

# Expected Outcome
Table should contain all the text I wrote, or an error or warning should be reported.

# Actual Output
The table entries are split on pipes disregarding the code backticks, and the resulting "extra columns" are silently discarded:

<img width="383" height="196" alt="Image" src="https://github.com/user-attachments/assets/f0e41f1e-a882-4887-a348-85cc5fea4ba3" />

This is apparently the specified behavior of GitHub-Flavored-Markdown tables, but it is certainly surprising. There should be at least a warning that content is being discarded.

**Comment by zachs18** [comment]
I agree that ideally there'd be a warning if a row of a table has more columns than the table itself (such that some are discarded).

Maybe something like 

```rs
warning: table row has too many columns
 --> src/lib.rs:1:41
  |
6 | //! | `code_with(|arg| arg)` |  `code_with(|arg| arg)` | three | four |
  |                                            ^ any content after this column divider is discarded
  |
  = help: this table only has 4 columns
  = help: to escape `|` characters in tables, add a `\` before them like `\|`.
  = note: `#[warn(rustdoc::oh_no_our_table_its_broken)]` on by default
```

**Comment by GuillaumeGomez** [comment]
That sounds like a bug in `pulldown-cmark` (the crate we use in rustdoc to handle markdown).

**Comment by GuillaumeGomez** [comment]
So apparently, like mentioned by @zachs18, the `|` character needs to be escaped because `pulldown-cmark` follows the [GFM spec](https://github.github.com/gfm/#tables-extension-). I agree that adding a warning here would be nice. Doing that.

EDIT: And it was also written in the first comment. I should re-read the full issue when I come back to it after some time. ^^'

**Comment by GuillaumeGomez** [comment]
Opened https://github.com/rust-lang/rust/pull/159583.
